subsys/bluetooth/mesh/shell/shell_ctl_cli.c

```diff
@@ -42,11 +42,17 @@ static int cmd_ctl_get(const struct shell *shell, size_t argc, char *argv[])
 
 static int ctl_set(const struct shell *shell, size_t argc, char *argv[], bool acked)
 {
-	uint16_t light = (uint16_t)strtol(argv[1], NULL, 0);
-	uint16_t temp = (uint16_t)strtol(argv[2], NULL, 0);
-	int16_t delta = (int16_t)strtol(argv[3], NULL, 0);
-	uint32_t time = (argc >= 5) ? (uint32_t)strtol(argv[4], NULL, 0) : 0;
-	uint32_t delay = (argc == 6) ? (uint32_t)strtol(argv[5], NULL, 0) : 0;
+	int err = 0;
+	uint16_t light = shell_strtoul(argv[1], 10, &err);
```

Comment:
This is already done in upstream, and it seems like it is not an issue.

Reply:
"Not an issue" as in it runs? If you put too big a number in to the shell, it will discard the higher bits. That is to me confusing behavior. If this is done with _signed_ integers, the result is implementation-defined, which is not ideal.

Reply:
Ye, if you input a too large number it would not be ideal. To prevent this would need an additional layer of arg checking i guess. By "not an issue" I meant that coverity did not complain during the upstream PR. I don't think typecasting would solve the issue with too large input values.

Reply:
I will approve the PR regardless of this. But I hope not we get a lot of coverity reports on this.